src/libraries/System.IO.Compression/tests/ZipArchive/zip_CreateTests.Comments.cs

```diff
@@ -9,40 +9,53 @@ namespace System.IO.Compression.Tests
-        [InlineData("ascii", "ascii!!!", "utf-8", "UÄäÖöÕõÜü")]
-        [InlineData("utf-8", "UÄäÖöÕõÜü", "ascii", "ascii!!!")]
-        [InlineData("ascii", "ascii!!!", "latin1", "LÄäÖöÕõÜü")]
-        [InlineData("latin1", "LÄäÖöÕõÜü", "ascii", "ascii!!!")]
-        [InlineData("utf-8", "UÄäÖöÕõÜü", "latin1", "LÄäÖöÕõÜü")]
-        [InlineData("latin1", "LÄäÖöÕõÜü", "utf-8", "UÄäÖöÕõÜü")]
-        public static void Create_ZipArchiveEntry_DifferentEncodings_FullName_And_Comment(string encodingName1, string text1, string encodingName2, string text2)
-        {
-            Encoding encoding1 = Encoding.GetEncoding(encodingName1);
-            Encoding encoding2 = Encoding.GetEncoding(encodingName2);
-            string entryName = encoding1.GetString(encoding1.GetBytes(text1));
-            string comment = encoding2.GetString(encoding2.GetBytes(text2));
+        [MemberData(nameof(Utf8Comment_Data))]
+        public static void Create_Comment_AsciiEntryName_NullEncoding(string originalComment, string expectedComment) =>
+            Create_Comment_EntryName_Encoding_Internal("file.txt", originalComment, expectedComment, null);
+
+        [Theory]
+        [MemberData(nameof(Utf8Comment_Data))]
+        public static void Create_Comment_AsciiEntryName_Utf8Encoding(string originalComment, string expectedComment) =>
+            Create_Comment_EntryName_Encoding_Internal("file.txt", originalComment, expectedComment, Encoding.UTF8);
 
-            var stream = new MemoryStream();
-            var testStream = new WrappedStream(stream, true, true, true, null);
+        [Theory]
+        [MemberData(nameof(Latin1Comment_Data))]
+        public static void Create_Comment_AsciiEntryName_Latin1Encoding(string originalComment, string expectedComment) =>
+            Create_Comment_EntryName_Encoding_Internal("file.txt", originalComment, expectedComment, Encoding.Latin1);
 
-            // Create with no encoding to autoselect it if one of the two strings is unicode
-            using (var zip = new ZipArchive(testStream, ZipArchiveMode.Create, leaveOpen: true))
+        [Theory]
+        [MemberData(nameof(Utf8Comment_Data))]
+        public static void Create_Comment_Utf8EntryName_NullEncoding(string originalComment, string expectedComment) =>
+            Create_Comment_EntryName_Encoding_Internal($"{SmileyEmoji}.txt", originalComment, expectedComment, null);
+
+        [Theory]
+        [MemberData(nameof(Utf8Comment_Data))]
+        public static void Create_Comment_Utf8EntryName_Utf8Encoding(string originalComment, string expectedComment) =>
+            Create_Comment_EntryName_Encoding_Internal($"{SmileyEmoji}.txt", originalComment, expectedComment, Encoding.UTF8);
+
+        [Theory]
+        [MemberData(nameof(Latin1Comment_Data))]
+        public static void Create_Comment_Utf8EntryName_Latin1Encoding(string originalComment, string expectedComment) =>
+            // Emoji not supported by latin1
```

Comment:
What happens if you pass an emoji with latin1 encoding? I assume it just encodes it as something meaningless. We should assert that the latin1-encoded-emoji still gets truncated.

Reply:
I'm not sure how I would assert that in the tests. I just wanted to make sure the string got truncated, even if the final outputted string, when viewed by the user, will be garbage where the emoji was saved.

Reply:
> I'm not sure how I would assert that in the tests.

On `Latin1Comment_Data` add something like this:
```
string latin1ExpectedALettersAndTruncatedEmoji = new string('a', ushort.MaxValue - 1) + SmileyEmoji[0];
string latin1OriginalALettersAndEmoji = new string('a', ushort.MaxValue - 1) + SmileyEmoji;
```
My point being that the truncation logic only runs when it makes sense (UTF8).